mlflow/sagemaker/__init__.py

```diff
@@ -1747,6 +1779,480 @@ def _does_model_exist(model_name, sage_client):
+                         process. It will not wait for the deployment process to complete;
+                         in this case, the caller is responsible for monitoring the health and
+                         status of the pending deployment via native SageMaker APIs or the AWS
+                         console.
+
+                       - timeout_seconds: If ``synchronous`` is ``True``, the deployment process
+                         will return after the specified number of seconds if no definitive result
+                         (success or failure) is achieved. Once the function returns, the caller is
+                         responsible for monitoring the health and status of the pending
+                         deployment using native SageMaker APIs or the AWS console. If
+                         ``synchronous`` is ``False``, this parameter is ignored.
+
+                       - vpc_config: A dictionary specifying the VPC configuration to use when
+                         creating the new SageMaker model associated with this application.
+                         The acceptable values for this parameter are identical to those of the
+                         ``VpcConfig`` parameter in the `SageMaker boto3 client's create_model
+                         method <https://boto3.readthedocs.io/en/latest/reference/services/sagemaker.html
+                         #SageMaker.Client.create_model>`_. For more information, see
+                         https://docs.aws.amazon.com/sagemaker/latest/dg/API_VpcConfig.html.
+
+                         .. code-block:: python
+                             :caption: Example
+
+                             import mlflow.sagemaker as mfs
+                             vpc_config = {
+                                             'SecurityGroupIds': [
+                                                 'sg-123456abc',
+                                             ],
+                                             'Subnets': [
+                                                 'subnet-123456abc',
+                                             ]
+                                          }
+                             client = mfs.SageMakerPlugin("sagemaker")
+                             client.create_deployment(..., config=dict(vpc_config=vpc_config))
+        """
+        final_config = self._default_deployment_config()
+        if config:
+            self._apply_custom_config(final_config, config)
+
+        app_name, flavor = deploy(
+            app_name=name,
+            model_uri=model_uri,
+            flavor=flavor,
+            execution_role_arn=final_config["execution_role_arn"],
+            assume_role_arn=final_config["assume_role_arn"],
+            bucket=final_config["bucket"],
+            image_url=final_config["image_url"],
+            region_name=final_config["region_name"],
+            mode=DEPLOYMENT_MODE_CREATE,
```

Comment:
On second thought, can we check if `mode` is specified in the `config` and, if it is, pass it in? If `mode` is not specified, we can fall back to `DEPLOYMENT_MODE_CREATE`.

There are a couple reasons I think we should do this:

1. Right now, if I try to use this method while specifying a deployment name that already exists, I get the following error message:

```
mlflow.exceptions.MlflowException: You are attempting to deploy an application with name: mydep2 in 'create' mode. However, an application with the same name already exists. If you want to update this application, deploy in 'add' or 'replace' mode.
```

This exception comes from `mlflow.sagemaker.deploy()` and would be inaccurate if the `replace` and `add` `mode`s couldn't be specified.

2. For users who are automating deployments and don't want to add logic to check whether or not a deployment exists before creating a new one, supporting `mode=replace` would provide users with a convenient single-method "create or update" capability.